middleware/haproxy/detectors-haproxy.tf

```diff
@@ -0,0 +1,144 @@
+resource "signalfx_detector" "heartbeat" {
+  name = "${join("", formatlist("[%s]", var.prefixes))}[${var.environment}] Haproxy heartbeat"
+
+  program_text = <<-EOF
+		from signalfx.detectors.not_reporting import not_reporting
+		signal = data('haproxy_session_current', filter=filter('aws_state', 'running') and filter('gcp_status', '*RUNNING}') and filter('azure_power_state', 'PowerState/running') and ${module.filter-tags.filter_custom}).publish('signal')
```

Comment:
this metric seems always reported but may be it is more "natural" to use the `haproxy_status` ?

Reply:
Yes that what I planned to choose at first but it is not a default metrics so I thought it was better to take this one which is a default one.